File: LeetCode/Easy/0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int minDepth(TreeNode root) {
        if(root == null){
            return 0;
        }

        int leftMin = minDepth(root.left);
        int rightMin = minDepth(root.right);
        if(root.left == null) return rightMin + 1;
        if(root.right == null) return leftMin + 1;
        int min = Math.min(leftMin, rightMin) + 1;
        return min;
        
    }
}

File: LeetCode/Easy/0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0

        queue = deque([(root, 1)])

        while queue:
            root, depth = queue.popleft() # 取出当前节点和深度
            if not root.left and not root.right:
                return depth

            if root.left:
                queue.append((root.left, depth + 1))

            if root.right:
                queue.append((root.right, depth + 1))


        